tinytuya/scanner.py

```diff
@@ -1175,6 +1284,11 @@ def tuyaLookup(deviceid):
     if not scantime:
         scantime = 0 if ip_scan_running else tinytuya.SCANTIME
 
+    client_bcast_addrs = get_ip_to_broadcast()
+    for bcast in client_bcast_addrs:
+        addr = client_bcast_addrs[bcast]
+        client_ip_broadcast_list[addr] = { 'broadcast': bcast }
+
```

Comment:
Would it be simpler to remove this logic and have `send_discovery_request()` build the broadcast list itself (assume we always want it to broadcast to all interfaces)? Or do you think we will want to alter / prune the list it would use?

Either is fine... just selfishly wondering if I could just call `scanner.send_discovery_request()` without worrying about sending it any arguments. 😋

Reply:
Yep, 2 reasons: so we only need to build the interface list/encrypt the payload/open the socket once, and so we can expand the list later if needed (i.e. we're using the single getmyIP() address and want to add any address we receive a broadcast on).

It shouldn't be hard to make the `send_discovery_request()` argument optional.